python/pyarrow/tests/test_dataset.py

```diff
@@ -1388,16 +1380,13 @@ def _create_dataset_all_types(tempdir, chunk_size=None):
     path = str(tempdir / "test_parquet_dataset_all_types")
 
     # write_to_dataset currently requires pandas
```

Comment:
This comment is no longer correct I think? (so that might also mean that all tests that use this helper no longer need the pandas mark)

Reply:
Oh yes, it was needed for `partition_cols` in the legacy path: https://github.com/apache/arrow/blob/87865b5a85c722ef7578aed4300e9d0b219c909c/python/pyarrow/parquet/core.py#L3516

Will remove.